packages/twenty-server/src/engine/core-modules/auth/services/reset-message-channel.service.ts

```diff
@@ -0,0 +1,78 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { Repository } from 'typeorm';
+
+import { DatabaseEventAction } from 'src/engine/api/graphql/graphql-query-runner/enums/database-event-action';
+import { ObjectMetadataEntity } from 'src/engine/metadata-modules/object-metadata/object-metadata.entity';
+import { WorkspaceEntityManager } from 'src/engine/twenty-orm/entity-manager/workspace-entity-manager';
+import { TwentyORMGlobalManager } from 'src/engine/twenty-orm/twenty-orm-global.manager';
+import { WorkspaceEventEmitter } from 'src/engine/workspace-event-emitter/workspace-event-emitter';
+import {
+  MessageChannelSyncStage,
+  MessageChannelWorkspaceEntity,
+} from 'src/modules/messaging/common/standard-objects/message-channel.workspace-entity';
+
+export type ResetMessageChannelsInput = {
+  workspaceId: string;
+  connectedAccountId: string;
+  manager: WorkspaceEntityManager;
+};
+
+@Injectable()
+export class ResetMessageChannelService {
+  constructor(
+    private readonly twentyORMGlobalManager: TwentyORMGlobalManager,
+    private readonly workspaceEventEmitter: WorkspaceEventEmitter,
+    @InjectRepository(ObjectMetadataEntity, 'metadata')
+    private readonly objectMetadataRepository: Repository<ObjectMetadataEntity>,
+  ) {}
+
+  async resetMessageChannels(input: ResetMessageChannelsInput): Promise<void> {
+    const { workspaceId, connectedAccountId, manager } = input;
+
+    const messageChannelRepository =
+      await this.twentyORMGlobalManager.getRepositoryForWorkspace<MessageChannelWorkspaceEntity>(
+        workspaceId,
+        'messageChannel',
+      );
+
+    const messageChannels = await messageChannelRepository.find({
+      where: { connectedAccountId },
+    });
+
+    const messageChannelUpdates = await messageChannelRepository.update(
```

Comment:
add a unit test to check that the messageChannelRepository is reset when we reconnect